### PR TITLE
Create show page

### PR DESCRIPTION
**Pull request by Crackey117**
added the paths necessary for a show page to work, updated the controller, created the show container component and show component (opened our browser to verify it worked), and then added tests (ensuring that it failed first) that now pass.

### CODE REVIEW

File: app/controllers/api/v1/movies_controller.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class Api::V1::MoviesController < ApplicationController
+  def index
+    movies = Movie.all
+    render json: movies 
+  end 
+
```
Comment (safang99) [inline app/controllers/api/v1/movies_controller.rb:6]: Looks like this is also adding in feature related to an index page that should be on their own branch and not the `create_show_page`

File: app/javascript/react/components/MoviesIndexPage.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  return (
+    <div>
+      <ul>
+        {movieListItems}
+      </ul>
+    </div> 
+  )
+}
+
+export default MoviesIndexPage
```
Comment (safang99) [inline app/javascript/react/components/MoviesIndexPage.js]: This whole file has nothing to do with the `show page` so it shouldnt be here

File: app/javascript/react/components/main.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import React from "react"
+import ReactDom from "react-dom"
+
+import App from "./App"
+
+ReactDom.render
+  (<App />, 
+  document.getElementById("app")
+)
```
Comment (safang99) [inline app/javascript/react/components/main.js]: This whole file can be deleted as we instead have a file in `app/javascript/packages/application.js` that will act as the pseudo `main.js`

File: app/models/movie.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+class Movie < ApplicationRecord
+  validates :title, presence: true
+  validates :summary, presence: true
+  validates :year, numericality: true
+end
+
+
+
+
```
Comment (safang99) [inline app/models/movie.rb]: empty lines 6-9 to be removed

File: config/routes.rb [hunk 5]
```diff
@@ -1,5 +1,18 @@
...
-  root 'homes#index'
   devise_for :users
+  
+  root 'homes#index'
+
+  get "/movies", to: "homes#index"
+
+  get "/movies/:id", to: "homes#index"
+
+
```
Comment (safang99) [inline config/routes.rb]: There dont need to be any empty lines between the routes from 4-8.  Also an extra empty line on 10

File: spec/controllers/api/v1/movies_controller_spec.rb [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+      expect(returned_json.length).to eq 1
+
+      expect(returned_json[0]["title"]).to eq "Star Wars"
+      expect(returned_json[0]["summary"]).to eq "Movie from childhood"
+      expect(returned_json[0]["year"]).to eq 1977
+
+    end
+  end
+end
```
Comment (safang99) [inline spec/controllers/api/v1/movies_controller_spec.rb:51]: This test should be added in on a  PR for the index feature

File: spec/controllers/api/v1/movies_show_controller_spec.rb [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+    expect(response.status).to eq 200
+    expect(response.content_type).to eq("application/json")
+   
+    expect(returned_json.length).to eq 6
+    expect(returned_json["title"]).to eq movie2.title
+    expect(returned_json["year"]).to eq movie2.year
+    expect(returned_json["summary"]).to eq movie2.summary
+    end
+  end
```
Comment (safang99) [inline spec/controllers/api/v1/movies_show_controller_spec.rb]: Excellent job here!

**Review by safang99 — CHANGES_REQUESTED** [review]
Looks like this was built of a branch that hasnt been merged yet

File: app/controllers/api/v1/movies_controller.rb (hunk 1 above)
Comment (Crackey117) [reply app/controllers/api/v1/movies_controller.rb:6]: repulled after the index branch merge was approved, should be good i think

File: app/javascript/react/components/MoviesIndexPage.js (hunk 2 above)
Comment (Crackey117) [reply app/javascript/react/components/MoviesIndexPage.js]: repulled after other PR was approved, this shouldnt be a problem now i think

File: app/javascript/react/components/main.js (hunk 3 above)
Comment (Crackey117) [reply app/javascript/react/components/main.js]: this was gone after re-pulling

File: spec/controllers/api/v1/movies_controller_spec.rb (hunk 6 above)
Comment (Crackey117) [reply spec/controllers/api/v1/movies_controller_spec.rb:51]: repulled after the index branch was merged, dont think this should be an issue

File: app/controllers/homes_controller.rb [hunk 8]
```diff
@@ -1,4 +1,8 @@
 class HomesController < ApplicationController
   def index
   end
+
+  def show
+  end
+
```
Comment (safang99) [inline app/controllers/homes_controller.rb]: Since this is never being used in `routes.rb` lets remove 4-7

File: config/routes.rb [hunk 9]
```diff
@@ -3,12 +3,12 @@
   resources :homes, only: [:index]
```
Comment (safang99) [inline config/routes.rb]: If we don't need this path lets remove line 3

**Review by safang99 — CHANGES_REQUESTED** [review]

File: app/controllers/homes_controller.rb (hunk 8 above)
Comment (Crackey117) [reply app/controllers/homes_controller.rb]: deleted it

File: config/routes.rb (hunk 9 above)
Comment (Crackey117) [reply config/routes.rb]: removed it

File: app/javascript/react/components/MovieShowContainer.js [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+      if (response.ok) {
+        return response.json();
+      } else {
+        let errorMessage = `${response.status} (${response.statusText})`,
+        error = new Error(errorMessage);
+        throw(error);
+      }
+      })
+      .then((responseBody) => {
+        debugger
```
Comment (dovakeen118) [inline app/javascript/react/components/MovieShowContainer.js]: DEBUGGER!

File: spec/controllers/api/v1/movies_show_controller_spec.rb [hunk 11]
```diff
@@ -0,0 +1,31 @@
+require "rails_helper"
```
Comment (dovakeen118) [inline spec/controllers/api/v1/movies_show_controller_spec.rb]: All tests for controller actions should be in the same file named after the controller you're testing. `spec/controllers/api/v1/movies_controller_spec.rb` will have all tests for all controller actions, keeping everything together

**Review by dovakeen118 — CHANGES_REQUESTED** [review]
Debugger was left in a component! Also want to combine the specs for a controller and its actions into one file

File: app/javascript/react/components/MovieShowContainer.js (hunk 10 above)
Comment (Crackey117) [reply app/javascript/react/components/MovieShowContainer.js]: whoops

File: spec/controllers/api/v1/movies_show_controller_spec.rb (hunk 11 above)
Comment (Crackey117) [reply spec/controllers/api/v1/movies_show_controller_spec.rb]: put them all in same file

**Pull request merged**